common/src/main/java/com/wynntils/features/user/WorldWaypointDistanceFeature.java

```diff
@@ -71,13 +71,17 @@ public void onRenderLevelPost(RenderLevelEvent.Post event) {
         if (Models.Compass.getCompassLocation().isEmpty()) return;
 
         Location location = Models.Compass.getCompassLocation().get();
-        Matrix4f projection = event.getProjectionMatrix().copy();
+        Matrix4f projection = new Matrix4f(event.getProjectionMatrix());
         Camera camera = event.getCamera();
         Vec3 cameraPos = camera.getPosition();
 
         // apply camera rotation
-        projection.multiply(Vector3f.XP.rotationDegrees(camera.getXRot()));
-        projection.multiply(Vector3f.YP.rotationDegrees(camera.getYRot() + 180.0F));
+        Vector3f xp = new Vector3f(1, 0, 0);
+        Vector3f yp = new Vector3f(0, 1, 0);
+        Quaternionf xRotation = new Quaternionf().rotationAxis((float) Math.toRadians(camera.getXRot()), xp);
+        Quaternionf yRotation = new Quaternionf().rotationAxis((float) Math.toRadians(camera.getYRot() + 180f), yp);
+        projection.mul(new Matrix4f().rotation(xRotation));
+        projection.mul(new Matrix4f().rotation(yRotation));
```

Comment:
I think it's probably 20 years since I forgot everything I ever knew about vector algebra. :-) It looks sane, but let me just ask: have you verified that the world waypoint seem to work? (It's more like if it's broken it's likely severly broken)

Reply:
Yes, I did. I even talked to the original creator of this code (as I failed to port a singular line of change). But now, it works as intended.